### PR TITLE
Minor doc updates

### PR DESCRIPTION
**Pull request by asears**
As I was going through the documentation, made a few minor improvements.

1. Fix some minor scan issues using markdownlint/markdownlint-cli
2. Add some missing howto's to the readme TOC.
3. Additions to example for setting up ruleset file in csproj.
4. Code block fencing

### CODE REVIEW

File: docs/HowToUpdateDocumentation.md [hunk 1]
```diff
@@ -1,8 +1,18 @@
 ﻿
 # How to: Update Documentation
 
+The following documentation for project in can be updated:
```
Comment (JosefPihrt) [inline docs/HowToUpdateDocumentation.md:4]: This sentence does not make sense to me.

File: docs/HowToUpdateDocumentation.md [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+
+```shell
+/docs/*.md
+/docs/cli/*.md
+```
+
+Documentation for project api in /docs/api is generated and should not be edited directly.
```
Comment (JosefPihrt) [inline docs/HowToUpdateDocumentation.md:11]:
```suggestion
Documentation for project api in `/docs/api` is generated and should not be edited directly.
```

File: docs/cli/analyze-command.md [hunk 3]
```diff
@@ -73,7 +73,7 @@ Defines project language.
 
 **`--msbuild-path`** <MSBUILD_PATH>
 
-Defines a path to MSBuild. This option must be specified if there are multiple locations of MSBuild (usually multiple installations of Visual Studio).
+Defines a path to MSBuild. This option must be specified if there are multiple locations of MSBuild.  This is usually required when multiple versions of Visual Studio are installed.
```
Comment (JosefPihrt) [inline docs/cli/analyze-command.md:76]:
```suggestion
Defines a path to MSBuild. This option must be specified if there are multiple locations of MSBuild. This is usually required when multiple versions of Visual Studio are installed.
```

**Review by JosefPihrt — CHANGES_REQUESTED** [review]
I just propose some minor improvements to your changes.

File: docs/HowToUpdateDocumentation.md (hunk 1 above)
Comment (asears) [reply docs/HowToUpdateDocumentation.md]: Me neither.

**Review by JosefPihrt — APPROVED** [review]

**Comment by JosefPihrt** [comment]
Thanks for the contribution!

I appreciate this PR because there are people that are complaining about the documentation but are not willing to make any improvements.

**Pull request merged**